src/App.js

```diff
@@ -14,11 +14,96 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    title: 'No goods selected',
+    list: [...goodsFromServer],
+    selectedGood: ['Jam'],
+  };
+
+  onAddProduct = (newTitle) => {
+    const { selectedGood } = this.state;
+
+    this.setState({
+      title: `${newTitle} is selected`,
+      selectedGood: [...selectedGood, newTitle],
+    });
+  };
+
+  onDeleteProduct = (product) => {
+    const index = this.state.selectedGood.indexOf(product);
+
+    if (index > -1) {
+      const { selectedGood: prev } = this.state;
+
+      prev.splice(index, 1);
```

Comment:
```suggestion
      prev.splice(index, 1);
```
better to use `filter` method for deleting